httpx/_compat.py

```diff
@@ -17,6 +18,23 @@
     except ImportError:
         brotli = None
 
+# Zstandard support is optional
+try:
+    import zstandard as zstd
+except (AttributeError, ImportError, ValueError):  # Defensive:
+    zstd = False
+else:
+    # The package 'zstandard' added the 'eof' property starting
+    # in v0.18.0 which we require to ensure a complete and
+    # valid zstd stream was fed into the ZstdDecoder.
+    # See: https://github.com/urllib3/urllib3/pull/2624
+    _zstd_version = tuple(
+        map(int, re.search(r"^([0-9]+)\.([0-9]+)", zstd.__version__).groups())  # type: ignore[union-attr]
+    )
+    if _zstd_version < (0, 18):  # Defensive:
+        zstd = False
```

Comment:
This is just out of my curiosity, feel free to not answer it.
Why do we need this code if we already specify zstandard>=0.18.0, wouldn't package manager guarantee it already?

Reply:
zstandard is an optional install. So if your `requirements.txt` just has `httpx` and not `httpx[zstd]`, and you have `zstandard==[old version]` somehow, this would prevent breakage.

Reply:
That's so smart. Thank you!